app/MarsRover/page.js

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+export default function MarsRover() {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    const fetchingMarsRover = async () => {
+      try {
+        const fetchData = await fetch(
+          "https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?sol=1000&api_key=DEMO_KEY"
+        );
+        const response = await fetchData.json();
+        setData(response.photos);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+    fetchingMarsRover();
+  }, []);
+
+  if (data.length === 0) {
+    return <div>Loading ...</div>;
```

Comment:
Is this the best way to see if it is loading? I think it is better if you set data to null at first so that if the API returns an empty array then this doesn't happen.

Reply:
Thanks for instruction, Now understood that empty array could also be the response, so one should not set data to empty array instead set it to null